### ISSUE TITLE
Discussion: Using an "importance" mask for loss?

### ISSUE THREAD

**Issue by Quasimondo** [phillipi/pix2pix#59]
This is more of a raw idea I hope to get some feedback on:

When training a pix2pix model I notice that usually there is no distinction being made between important and not-important details. So for example the model tries just as hard to reproduce the background of a scene than the foreground. So when I try to train faces the details usually never get beyond a certain quality level which I attribute to the model having to balance its available resources across all features.

So I wonder if I could tell the model to learn more of the important and less of the unimportant features by passing a 3rd map along with my training pairs which would be a kind of "heat map" that gives an importance value for every pixel. So in the case of a face I might want to give special emphasis on the eyes, the nose and the mouth, whereas the hair might be of less importance. I'm thinking of using this map as a multiplier for the loss before it gets sent into the backward() pass. In theory the loss in areas that are not interesting would become 0 or very small.

So my question is - does this make any sense? Of course I could just mask my training data, but this might also lead to unwanted discolorization. Or maybe there is a much simpler way of achieving this that I overlooked.

**Comment by junyanz** [comment]
I think this is a great idea. As you said, you can multiply the loss byyou mask. I guess there might be automatic way to get the mask for face data (such as face detection, or face alignment)
